File: app/src/main/java/edu/gatech/cs2340/spacetraderredux/Model/Planet.java
package edu.gatech.cs2340.spacetraderredux.Model;

import java.util.Random;

public class Planet {

    private final PlanetName name;
    private final Resource resource;
    private final TechLevel techLevel;

    public Planet(Random seed) {
        this.name = PlanetName.values()[seed.nextInt(PlanetName.values().length)];
        this.resource = Resource.values()[seed.nextInt(Resource.values().length)];
        this.techLevel = TechLevel.values()[seed.nextInt(TechLevel.values().length)];
    }

    public PlanetName getName() {
        return name;
    }

    public Resource getResource() {
        return resource;
    }

    public TechLevel getTechLevel() {
        return techLevel;
    }

    @Override
    public String toString() {
        return "Planet " + name.name() + " with resource: " + resource.name() + " and tech level: " + techLevel.name();
    }
}

File: app/src/main/java/edu/gatech/cs2340/spacetraderredux/Model/Planet.kt
package edu.gatech.cs2340.spacetraderredux.Model

import java.util.Random

class Planet(rand: Random) {

    val name: PlanetName
    val resource: Resource
    val techLevel: TechLevel

    init {
        this.name = PlanetName.values()[rand.nextInt(PlanetName.values().size)]
        this.resource = Resource.values()[rand.nextInt(Resource.values().size)]
        this.techLevel = TechLevel.values()[rand.nextInt(TechLevel.values().size)]
    }

    override fun toString(): String {
        return "Planet " + name.name + " with resource: " + resource.name + " and tech level: " + techLevel.name
    }
}
